components/prism-odin.js

```diff
@@ -0,0 +1,61 @@
+(function (Prism) {
+	var escapes = /\\(?:["'\\abefnrtv]|0[0-7]{2}|U[\dA-Fa-f]{6}|u[\dA-Fa-f]{4}|x[\dA-Fa-f]{2})/.source;
+
+	function withEscapes(pattern, flags) {
+		return RegExp(pattern.replace(/<escapes>/g, escapes), flags);
+	}
+
+	Prism.languages.odin = {
+		/**
+		 * The current implementation of multiline comments doesn't support nesting.
+		 */
+		'comment': [
+			{
+				pattern: /\/\*[\s\S]*?(?:\*\/|$)/,
+				greedy: true
+			},
+			{
+				pattern: /#![^\n\r]*/,
+				greedy: true
+			},
+			{
+				pattern: /\/\/[^\n\r]*/,
+				greedy: true
+			}
+		],
+
+		/**
+		 * Should be found before strings because of '"'"- and '`'`-like sequences.
+		 */
+		'char': {
+			pattern: withEscapes(/'(?:<escapes>|[^\n\r'\\])'/.source),
+			inside: {
+				'symbol': withEscapes(/<escapes>/.source)
+			}
+		},
+
+		'string': [
+			{
+				pattern: /`[^`]*`/,
+				greedy: true
+			},
+			{
+				pattern: withEscapes(/"(?:<escapes>|[^\n\r"\\])*"/.source),
```

Comment:
Same as with `char`, handling escapes explicitly has downsides.

```suggestion
				pattern: /"(?:\\.|[^\n\r"\\])*"/,
```